pkg/deploy/dev-workspace-config/dev_workspace_config.go

```diff
@@ -180,6 +193,78 @@ func updateProjectCloneConfig(devEnvironments *chev2.CheClusterDevEnvironments,
 	workspaceConfig.ProjectCloneConfig.Resources = cheResourcesToCoreV1Resources(container.Resources)
 }
 
+// Merges the given container security context with the container security context required for the container builds SCC
+// and returns the resulting container security context
+func mergeContainerSecurityContextForContainerBuilds(containerSecurityContext *corev1.SecurityContext) *corev1.SecurityContext {
+	addCapabilities, dropCapabilities := mergeSecurityContextCapabilities(constants.DefaultWorkspaceContainerSecurityContext.Capabilities, containerSecurityContext.Capabilities)
+
+	resultingContainerSecurityContext := containerSecurityContext.DeepCopy()
+	resultingContainerSecurityContext.Capabilities = &corev1.Capabilities{
+		Add:  addCapabilities,
+		Drop: dropCapabilities,
+	}
+	resultingContainerSecurityContext.AllowPrivilegeEscalation = constants.DefaultWorkspaceContainerSecurityContext.AllowPrivilegeEscalation
+
+	return resultingContainerSecurityContext
+}
+
+// Merges the capabilities of from into to, removing any dropped capabilities in to that are being added in from
+// and removing any added capabilities in to that are being dropped in from.
+func mergeSecurityContextCapabilities(from, to *corev1.Capabilities) (addCapabilities, dropCapabilities []corev1.Capability) {
+	// Returns a map of corev1.capability -> bool to be used as a set, ensuring each capability is unique
+	getCapabilitiesSet := func(capabilities []corev1.Capability) map[corev1.Capability]bool {
+		capabilityBoolMap := map[corev1.Capability]bool{}
+		for _, capability := range capabilities {
+			capabilityBoolMap[capability] = true
+		}
+		return capabilityBoolMap
+	}
+
+	// Get Add Capabilities that will be merged into
+	toAddCapabilitiesMap := getCapabilitiesSet(to.Add)
+
+	// Get Drop Capabilities that will be merged into
+	toDropCapabilitiesMap := getCapabilitiesSet(to.Drop)
+
+	// Get Add Capabilities that will be merged from
+	fromAddCapabilitiesMap := getCapabilitiesSet(from.Add)
+
+	// Get Drop Capabilities that will be merged from
+	fromDropCapabilitiesMap := getCapabilitiesSet(from.Drop)
+
+	// Remove add capabilities that are being dropped
+	for dropCapability := range fromDropCapabilitiesMap {
```

Comment:
Note: for this PR, we don't _need_ to be removing add capabilities that are dropped in `from`, since `from` is `constants.DefaultWorkspaceContainerSecurityContext.Capabilities` which does not drop any capabilities.

However, I figured it'd make more sense to handle all cases, even if they aren't needed.